src/blockmap.jl

```diff
@@ -83,7 +92,9 @@ function Base.hcat(As::Union{LinearMap,UniformScaling,AbstractVecOrMat}...)
             break
         end
     end
-    nrows == -1 && throw(ArgumentError("hcat of only UniformScaling objects cannot determine the linear map size"))
+    errmsg = "hcat of only UniformScaling objects cannot determine the linear map size"
+    nrows == -1 && throw(ArgumentError(errmsg))
+    # does this mean we still have type piracy? I vagely remember this being discussed
```

Comment:
Ok great. Annoying that this does not work for `cat`. Looking into the implementation of `cat` in Base, LinearAlgebra and SparseArrays, it seems like a complete mess. Something like
```
cat(rand(Float32, (2,2)), rand(Float64, (3,3)); dims = (1,2))
```
goes via SparseArrays. Anyway, looking at SparseArrays, it seems like we might want to overload `Base._cat` instead of `Base.cat`?

Reply:
Also, for the above, it means that it is impossible to trigger the error message?

Reply:
> it seems like we might want to overload Base._cat instead of Base.cat

If that does the job, that seems fine to me.

> Also, for the above, it means that it is impossible to trigger the error message?

Yes, indeed. Good point!

Reply:
too bad that `?Base._cat` returns nothing useful.  sigh.

Reply:
This was a very helpful remark, Jutho! Overloading `Base._cat` instead of `Base.cat` allows to relax the requirement of "a LinearMap object among the first 8 arguments"! I'll include that in #114. 🎉

Reply:
Oh yes, I completely forgot about this, I was planning to change that myself, but if you can do this, way better :-)